### ISSUE TITLE
z-index problem

### ISSUE THREAD

**Issue by Bendico** [n33/jquery.dropotron#8]
I'm using skel panels. My dropotrons are obscured by the parent #nav, and have found this is because the ul's for dropotrons are detached from the parent and inserted in a div above the #nav in the DOM. Hence, using z-index is of no use, because the panel containing the dropotron would then effectively hide the entire #nav bar, which is an overlay panel.

If I try to override the 'detach' parameter for dropotron, z-indexing works fine, but the positioning is wrong. It appears that the positioning calculation does not take into account that the dropotron is now a child of the parent menu item and not the page panel.

**Comment by n33** [comment]
Have you tried changing the `baseZIndex` used by Dropotron?

**Comment by Bendico** [comment]
Yes. The problem with this, is that when the dropmenu is detached from the nav, it is placed in an entirely different skel panel which is z-indexed "under" the nav panel, which is an overlay panel. Since the panels are entirely separate divs with z-index and positioning, z-index of the dropmenu will only affect elements within that panel. In other words, to have the dropmenu on top, I will have to z-index the page panel above the nav panel, effectively hiding the whole menu entirely. If I could control the detachment, it would solve the problem, is it possible to tell dropotron where in the DOM to attach itself?

**Comment by n33** [comment]
Not without modifying Dropotron itself, but if you're okay with that, this bit of code (line 534 of the unminified source) controls how that works:

```js
// Detach the menu and move it to the end of the <body> element.
	$menu
		.detach()
		.appendTo($body);
```

Changing `$body` to a different element/selector (eg. `$('#myelement')`) ought to do the trick.

**Comment by Bendico** [comment]
Thanks a million, problem solved! I actually elaborated a little bit and added a setting 'attachSelector' with default value 'body', then replaced the harcoded ('body') in line 97 with (settings.attachSelector). So now I can customize the attachment from the initializer, if needed, otherwise it will default to body as before.

I also figured out that it is the skel-panels framework that actually causes the problem, because it reorganizes things in the DOM after dropotron has been initialized. So my solution here was to append the dropotron to the #nav element and use a ul:not(.dropotron) css selector to get the styling right.

BTW, the project I am working on is customizing the TXT Starter Kit for Umbraco CMS, which I believe is originally your TXT by HTML5UP, right? However, the submenus are not implemented in the Starter Kit, and that is what I have accomplished now actually, thanks to your help:-)

Would you consider making my attachSelector setting an official part of the dropotron code? It would mean I don't have to synchronize any updates, and maybe someone else could even find use for it? (I'm thinking particularly of Umbraco and skel-panels users.)

**Comment by n33** [comment]
Sure, I'll look into doing that in the next release.
